File: HandSock-Server/src/main/java/com/server/handsock/sockets/handler/SysConfigHandler.java
package com.server.handsock.sockets.handler;


import com.corundumstudio.socketio.AckRequest;
import com.server.handsock.console.HandleResults;
import com.server.handsock.services.ClientService;
import com.corundumstudio.socketio.SocketIOClient;
import com.server.handsock.admin.service.ServerSystemService;

import java.util.Map;

public class SysConfigHandler {

    private final Map<String, Object> service;
    private final Map<String, Object> clientServiceList;
    private final Map<String, Object> serverServiceList;
    private final HandleResults handleResults = new HandleResults();

    public SysConfigHandler(Map<String, Object> service, Map<String, Object> clientServiceList, Map<String, Object> serverServiceList) {
        this.service = service;
        this.clientServiceList = clientServiceList;
        this.serverServiceList = serverServiceList;
    }

    private boolean isValidClient(SocketIOClient client) {
        ClientService clientService = new ClientService(service, clientServiceList);
        return clientService.validClientToken(client) && clientService.getIsAdmin(client);
    }

    public void handleSetSystemTaboo(SocketIOClient client, Map<String, Object> data, AckRequest ackSender) {
        ClientService clientService = new ClientService(service, clientServiceList);
        if (isValidClient(client)) {
            try {
                ServerSystemService serverSystemService = (ServerSystemService) serverServiceList.get("serverSystemService");
                String value = clientService.getClientData(data, "value");
                ackSender.sendAckData(serverSystemService.setSystemTabooStatus(value));
            } catch (Exception e) {
                clientService.handleException(e, ackSender);
            }
        } else ackSender.sendAckData(handleResults.handleResultByCode(403, null, "禁止访问"));
    }

    public void handleSetSystemUpload(SocketIOClient client, Map<String, Object> data, AckRequest ackSender) {
        ClientService clientService = new ClientService(service, clientServiceList);
        if (isValidClient(client)) {
            try {
                ServerSystemService serverSystemService = (ServerSystemService) serverServiceList.get("serverSystemService");
                String value = clientService.getClientData(data, "value");
                ackSender.sendAckData(serverSystemService.setSystemUploadStatus(value));
            } catch (Exception e) {
                clientService.handleException(e, ackSender);
            }
        } else ackSender.sendAckData(handleResults.handleResultByCode(403, null, "禁止访问"));
    }

    public void handleSetSystemRegister(SocketIOClient client, Map<String, Object> data, AckRequest ackSender) {
        ClientService clientService = new ClientService(service, clientServiceList);
        if (isValidClient(client)) {
            try {
                ServerSystemService serverSystemService = (ServerSystemService) serverServiceList.get("systemService");
                String value = clientService.getClientData(data, "value");
                ackSender.sendAckData(serverSystemService.setSystemRegisterStatus(value));
            } catch (Exception e) {
                clientService.handleException(e, ackSender);
            }
        } else ackSender.sendAckData(handleResults.handleResultByCode(403, null, "禁止访问"));
    }

    public void handleSetSystemPlaylist(SocketIOClient client, Map<String, Object> data, AckRequest ackSender) {
        ClientService clientService = new ClientService(service, clientServiceList);
        if (isValidClient(client)) {
            try {
                ServerSystemService serverSystemService = (ServerSystemService) serverServiceList.get("serverSystemService");
                String value = clientService.getClientData(data, "value");
                ackSender.sendAckData(serverSystemService.setSystemPlaylistValue(value));
            } catch (Exception e) {
                clientService.handleException(e, ackSender);
            }
        } else ackSender.sendAckData(handleResults.handleResultByCode(403, null, "禁止访问"));
    }

    public void handleGetSystemPlaylist(SocketIOClient client, AckRequest ackSender) {
        ClientService clientService = new ClientService(service, clientServiceList);
        if (isValidClient(client)) {
            try {
                ServerSystemService serverSystemService = (ServerSystemService) serverServiceList.get("serverSystemService");
                ackSender.sendAckData(serverSystemService.getSystemKeyConfig("playlist"));
            } catch (Exception e) {
                clientService.handleException(e, ackSender);
            }
        } else ackSender.sendAckData(handleResults.handleResultByCode(403, null, "禁止访问"));
    }

    public void handleGetAllSystemConfig(SocketIOClient client, AckRequest ackSender) {
        ClientService clientService = new ClientService(service, clientServiceList);
        if (isValidClient(client)) {
            try {
                ServerSystemService serverSystemService = (ServerSystemService) serverServiceList.get("serverSystemService");
                ackSender.sendAckData(serverSystemService.getAllSystemConfig());
            } catch (Exception e) {
                clientService.handleException(e, ackSender);
            }
        } else ackSender.sendAckData(handleResults.handleResultByCode(403, null, "禁止访问"));
    }
}

File: HandSock-Server/src/main/java/com/server/handsock/sockets/handler/SysConfigHandler.kt
package com.server.handsock.sockets.handler

import com.corundumstudio.socketio.AckRequest
import com.corundumstudio.socketio.SocketIOClient
import com.server.handsock.admin.service.ServerSystemService
import com.server.handsock.services.AuthService
import com.server.handsock.services.ClientService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class SysConfigHandler @Autowired constructor(
    private val authService: AuthService,
    private val clientService: ClientService,
    private val serverSystemService: ServerSystemService
) {
    fun handleSetSystemTaboo(client: SocketIOClient, data: Map<String?, Any>, ackSender: AckRequest) {
        ackSender.sendAckData(authService.validAdminStatusBySocket(client) {
            val value = clientService.getClientData(data, "value")
            serverSystemService.setSystemTabooStatus(value)
        })
    }

    fun handleSetSystemUpload(client: SocketIOClient, data: Map<String?, Any>, ackSender: AckRequest) {
        ackSender.sendAckData(authService.validAdminStatusBySocket(client) {
            val value = clientService.getClientData(data, "value")
            serverSystemService.setSystemUploadStatus(value)
        })
    }

    fun handleSetSystemRegister(client: SocketIOClient, data: Map<String?, Any>, ackSender: AckRequest) {
        ackSender.sendAckData(authService.validAdminStatusBySocket(client) {
            val value = clientService.getClientData(data, "value")
            serverSystemService.setSystemRegisterStatus(value)
        })
    }

    fun handleSetSystemConfigValue(client: SocketIOClient, data: Map<String?, Any>, ackSender: AckRequest) {
        ackSender.sendAckData(authService.validAdminStatusBySocket(client) {
            val name = clientService.getClientData(data, "name")
            val value = clientService.getClientData(data, "value")
            serverSystemService.setSystemConfigValue(
                name = name,
                value = value
            )
        })
    }

    fun handleGetSystemPlaylist(client: SocketIOClient, ackSender: AckRequest) {
        ackSender.sendAckData(authService.validAdminStatusBySocket(client) {
            serverSystemService.getSystemKeyConfig("playlist")
        })
    }

    fun handleGetAllSystemConfig(client: SocketIOClient, ackSender: AckRequest) {
        ackSender.sendAckData(authService.validAdminStatusBySocket(client) {
            serverSystemService.allSystemConfig
        })
    }
}
